### PR TITLE
feat: owl object detector

### PR DESCRIPTION
**Pull request by jarulraj**
👋 Thanks for submitting a Pull Request to EVA DB!

🙌 We want to make contributing to EVA DB as easy and transparent as possible. Here are a few tips to get you started:

- 🔍 Search existing EVA DB [PRs](https://github.com/georgia-tech-db/eva/pulls) to see if a similar PR already exists.
- 🔗 Link this PR to a EVA DB [issue](https://github.com/georgia-tech-db/eva/issues) to help us understand what bug fix or feature is being implemented.
- 📈 Provide before and after profiling results to help us quantify the improvement your PR provides (if applicable).

👉 Please see our ✅ [Contributing Guide](https://evadb.readthedocs.io/en/stable/source/contribute/index.html) for more details.

### CODE REVIEW

File: eva/udfs/owl_object_detector.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+from eva.catalog.catalog_type import NdArrayType
+from eva.udfs.abstract.abstract_udf import AbstractUDF
+from transformers import OwlViTProcessor, OwlViTForObjectDetection
+from eva.utils.logging_manager import logger
+from eva.udfs.gpu_compatible import GPUCompatible
+from eva.udfs.decorators.decorators import forward, setup
+from eva.udfs.decorators.io_descriptors.data_types import PandasDataframe
+
+try:
```
Comment (AlexanderCui001228) [inline eva/udfs/owl_object_detector.py:28]: is there a reason for using a try-catch block here?

File: eva/udfs/owl_object_detector.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        self.device = device
+        return self
+
+    @setup(cachable=True, udf_type="object_detection", batchable=True)
+    def setup(self, threshold=0.3):
+        logging.getLogger("hf_oneshot").setLevel(logging.CRITICAL)  # yolov5
+        self.threshold = threshold
+        self.processor = OwlViTProcessor.from_pretrained("google/owlvit-base-patch32")
+        self.model = OwlViTForObjectDetection.from_pretrained("google/owlvit-base-patch32")
+        # self.model = torch.hub.load("ultralytics/yolov5", "yolov5s", verbose=False)
```
Comment (AlexanderCui001228) [inline eva/udfs/owl_object_detector.py:60]: deadcode

File: eva/udfs/owl_object_detector.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        """
+        # Stacking all frames, and changing to numpy
+        # because of yolov5 error with Tensors
+        print("Does this work?")
+        outcome = []
+        # Convert to HWC
+        # https://github.com/ultralytics/yolov5/blob/3e55763d45f9c5f8217e4dad5ba1e6c1f42e3bf8/models/common.py#L658
+        frames = np.ravel(frames.to_numpy())
+        frames = torch.permute(frames, (0, 2, 3, 1))
+        logger.warn("Does this work?1")
```
Comment (AlexanderCui001228) [inline eva/udfs/owl_object_detector.py:102]: These logger.warn lines merely check if each line works during initial debugging. Perhaps they should be deleted?

**Comment by jarulraj** [comment]
No longer needed due to https://github.com/georgia-tech-db/eva/pull/718

**Pull request closed without merging**